### PR TITLE
feat: add index on builds.created

### PR DESCRIPTION
**Pull request by dtanner**
Used for reporting - e.g. `where created > $some_epoch` to quickly and cheaply limit querying recent build metrics.

### CODE REVIEW

File: database/postgres/ddl/build.go [hunk 1]
```diff
@@ -61,5 +61,11 @@ CREATE INDEX
 IF NOT EXISTS
 builds_status
 ON builds (status);
+`
```
Comment (wass3r) [inline database/postgres/ddl/build.go:64]: super picky but for consistency, can we add blank line after this and add a comment before the new query?

File: database/postgres/postgres.go [hunk 2]
```diff
@@ -305,5 +305,10 @@ func createIndexes(c *client) error {
 		return fmt.Errorf("unable to create workers_hostname_address index for the %s table: %v", constants.TableWorker, err)
 	}
 
+	err = c.Postgres.Exec(ddl.CreateBuildCreatedIndex).Error
```
Comment (wass3r) [inline database/postgres/postgres.go:308]: missing comment before this for consistency

File: database/sqlite/ddl/build.go [hunk 3]
```diff
@@ -61,5 +61,12 @@ CREATE INDEX
 IF NOT EXISTS
 builds_status
 ON builds (status);
+`
+
+	CreateBuildCreatedIndex = `
```
Comment (wass3r) [inline database/sqlite/ddl/build.go:66]: missing comment right before this new query.

File: database/sqlite/sqlite.go [hunk 4]
```diff
@@ -256,6 +256,11 @@ func createIndexes(c *client) error {
 		return fmt.Errorf("unable to create builds_status index for the %s table: %v", constants.TableBuild, err)
 	}
 
+	err = c.Sqlite.Exec(ddl.CreateBuildCreatedIndex).Error
```
Comment (wass3r) [inline database/sqlite/sqlite.go:259]: missing comment before index creation.

additionally, i noticed between here and the postgres addition that the index is inserted in different spots. in the postgres version it was inserted at the end of the indexes and here after the other index on the build table. if i had to pick, i like the approach here better, but feel like it should be consistent between the two no matter where it happens.

**Review by wass3r — CHANGES_REQUESTED** [review]
thanks for the additions! left some comments. super minor tweaks - mostly for consistency.

**Comment by dtanner** [comment]
i think all your comments were addressed @wass3r

**Review by wass3r — APPROVED** [review]

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**